sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -182,309 +182,61 @@ private static String namespaceOf(ClassForDisplay clazz) {
-     */
-    ItemBuilder addIfNotNull(String key, @Nullable Duration value);
-
-    /**
-     * Register the given duration display data if the value is different than the specified
-     * default.
-     *
-     * @see DisplayData.Builder#add(String, Duration)
-     */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable Duration value, @Nullable Duration defaultValue);
-
-    /**
-     * Register the given class display data. The metadata item will be registered with type
-     * {@link DisplayData.Type#JAVA_CLASS}, and is identified by the specified key and namespace
-     * from the current transform or component.
-     */
-    ItemBuilder add(String key, Class<?> value);
-
-    /**
-     * Register the given class display data. The metadata item will be registered with type
-     * {@link DisplayData.Type#JAVA_CLASS}, and is identified by the specified key and namespace
-     * from the current transform or component.
-     */
-    ItemBuilder add(String key, ClassForDisplay value);
-
-    /**
-     * Register the given class display data if the value is not null.
-     *
-     * @see DisplayData.Builder#add(String, Class)
-     */
-    ItemBuilder addIfNotNull(String key, @Nullable Class<?> value);
-
-    /**
-     * Register the given class display data if the value is not null.
-     *
-     * @see DisplayData.Builder#add(String, ClassForDisplay)
-     */
-    ItemBuilder addIfNotNull(String key, @Nullable ClassForDisplay value);
-
-    /**
-     * Register the given class display data if the value is different than the specified default.
-     *
-     * @see DisplayData.Builder#add(String, Class)
-     */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable Class<?> value, @Nullable Class<?> defaultValue);
-
-    /**
-     * Register the given class display data if the value is different than the specified default.
-     *
-     * @see DisplayData.Builder#add(String, ClassForDisplay)
-     */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable ClassForDisplay value, @Nullable ClassForDisplay defaultValue);
-  /**
-   * Register the given display data with the specified type.
-   *
-   * <p> The added display data is identified by the specified key and namespace from the current
-   * transform or component.
-   *
-   * @throws ClassCastException if the value cannot be safely cast to the specified type.
-   *
-   * @see DisplayData#inferType(Object)
-   */
-    ItemBuilder add(String key, Type type, Object value);
+    <T> Builder addIfNotDefault(Item<T> item, @Nullable T defaultValue);
   }
 
   /**
-   * Utility to append optional fields to display data, or register additional display data
-   * items.
+   * A display data item. DisplayData items are registered via {@link DisplayData.Builder#add}
+   * within {@link HasDisplayData#populateDisplayData} implementations. Each metadata item is
+   * uniquely identified by the specified key and namespace generated from the registering
```

Comment:
Calling `super.populateDisplayData()` will register the display data with the namespace of the child class. A bit confusing. I'm not convinced yet what the "correct" behavior should be: for a base class registering display data, does the base class "own" that data (and claim the namespace), or is the base class an implementation detail? I'm leaning towards the second.

Here's another go at clarifying the javadoc. I think it's still a bit confusing; I'll think about it more:

    /**
     * A display data item. {@link Item Items} are registered via {@link DisplayData.Builder#add}
     * within {@link HasDisplayData#populateDisplayData} implementations. Each metadata item is
     * uniquely identified by the specified key, and a namespace derived from the
     * {@link HasDisplayData} component.
     */